File: frontend/src/types/product.ts
export interface ProductImage {
  id: string
  url: string
  altText: string
  sortOrder: number
  isPrimary: boolean
}

export interface VariantOptionValue {
  id: string
  value: string
  sortOrder: number
}

export interface VariantOptionType {
  id: string
  name: string
  sortOrder: number
  values: VariantOptionValue[]
}

export interface ProductVariant {
  id: string
  sku: string
  name: string
  price: number
  comparePrice: number | null
  stock: number
  stockReserved: number
  isActive: boolean
  sortOrder: number
  optionValues: VariantOptionValue[]
}

export interface ProductListDto {
  id: string
  name: string
  slug: string
  brand: string | null
  basePrice: number
  comparePrice: number | null
  stock: number
  isFeatured: boolean
  primaryImage: string | null
  categoryName: string | null
  genero: string | null
  createdAt: string
}

export interface Product {
  id: string
  name: string
  slug: string
  description: string
  shortDescription: string
  sku: string
  brand: string | null
  basePrice: number
  comparePrice: number | null
  stock: number
  status: 'DRAFT' | 'ACTIVE' | 'ARCHIVED'
  isFeatured: boolean
  genero: string | null
  images: ProductImage[]
  category: { id: string; name: string; slug: string } | null
  variants: ProductVariant[]
  optionTypes: VariantOptionType[]
  createdAt: string
}

File: backend/src/main/java/com/ecommerce/catalog/entity/Product.java
package com.ecommerce.catalog.entity;

import com.ecommerce.auth.entity.User;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "short_description")
    private String shortDescription;

    @Column(unique = true)
    private String sku;

    private String brand;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Status status = Status.DRAFT;

    @Column(name = "is_featured")
    @Builder.Default
    private boolean isFeatured = false;

    @Column(name = "meta_title")
    private String metaTitle;

    @Column(name = "meta_description")
    private String metaDescription;

    @Column(columnDefinition = "TEXT[]")
    private String[] tags;

    @Column(name = "weight_grams")
    private Integer weightGrams;

    @Column(name = "base_price", nullable = false, precision = 12, scale = 2)
    @Builder.Default
    private BigDecimal basePrice = BigDecimal.ZERO;

    @Column(name = "compare_price", precision = 12, scale = 2)
    private BigDecimal comparePrice;

    @Column(nullable = false)
    @Builder.Default
    private int stock = 0;

    @CreatedDate
    @Column(name = "created_at", updatable = false, nullable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "genero")
    private String genero;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    public enum Status {
        DRAFT, ACTIVE, ARCHIVED
    }
}
